frontend/src/components/VestingTable.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { useState } from 'react';
+import { CONTRACT_ADDR } from '../lib/config';
+import axios from 'axios';
+import { useAccount } from '@starknet-react/core';
+import { useQuery } from '@tanstack/react-query';
+import { BASE_API_URL } from "../lib/config";
+
+interface VestingEvent {
+    amount: number;
+    claimable_at: number | null;
+    is_claimable: boolean;
+    is_claimed: boolean;
+}
+
+const ITEMS_PER_PAGE = 10; // Items per page
+
+const VestingTable: React.FC = () => {
+    const { address } = useAccount();
+    const [currentPage, setCurrentPage] = useState(1); // Track the current page
+
+    const { data: events = [], isLoading, error } = useQuery({
+        queryKey: ['vesting-events', address],
+        queryFn: async () => {
+            if (!address) {
+                return [];
+            }
+            const response = await axios.get(`${BASE_API_URL}vesting-events`, {
+                params: {
+                    contract: CONTRACT_ADDR,
+                    address: address,
+                }
+            });
+            return response.data;
+        },
+        enabled: !!address,  // Only fetch if the address is available
+        retry: false,        // Disable retries for this query
+    });
+
+    // Pagination logic
+    const startIndex = (currentPage - 1) * ITEMS_PER_PAGE;
+    const paginatedEvents = events.slice(startIndex, startIndex + ITEMS_PER_PAGE);
+    const totalPages = Math.ceil(events.length / ITEMS_PER_PAGE);
+
+    const handleNextPage = () => {
+        if (currentPage < totalPages) {
+            setCurrentPage(currentPage + 1);
+        }
+    };
+
+    const handlePreviousPage = () => {
+        if (currentPage > 1) {
+            setCurrentPage(currentPage - 1);
+        }
+    };
+
+    if (isLoading) {
+        return <div>Loading...</div>;
+    }
+
+    if (error) {
+        return <p>{(error as Error).message || 'Failed to load vesting events.'}</p>;
+    }
+
+    return (
+        <div>
+            {/* Title */}
+            <div className="flex w-full flex-grow pb-4 text-2xl font-bold">Vesting Event</div>
+
+            {/* Table */}
```

Comment:
In the review for transfer status, I pointed out that that was the third table-like entity on the page that looks different. This is the fourth table-like entity on the page now – and yet it looks different from all the others. Please make it look consistent with at least one of the others, preferably the Your Stakes I think. (But your call.)